### PR TITLE
Fix parity_netPeers and add more tests

### PR DESCRIPTION
**Pull request by marcindsobczak**
I needed to add IP2PProtocolHandler to be able to substitute Caps in ParityModuleTests. We recognized, that it's a good moment to clean up a little and move whole capability-connected content from IProtocolHandler to this new IP2PProtocolHandler and clean up in classes, which were unnecessarily implementing it.

Added:
* test for empty ActivePeers
* test for null ActivePeers
* test for empty Caps

### CODE REVIEW

File: src/Nethermind/Nethermind.JsonRpc.Test/Modules/ParityModuleTests.cs [hunk 1]
```diff
@@ -52,7 +53,9 @@ namespace Nethermind.JsonRpc.Test.Modules
     [TestFixture]
     public class ParityModuleTests
     {
-        private IParityModule _parityModule;
+        private IParityModule _parityModuleA;
+        private IParityModule _parityModuleB;
+        private IParityModule _parityModuleC;
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.JsonRpc.Test/Modules/ParityModuleTests.cs:58]: parity module B & C are used only in 1 test, they can be constructed there as variables

File: src/Nethermind/Nethermind.Network/P2P/P2PProtocolHandler.cs [hunk 2]
```diff
@@ -30,11 +30,14 @@
 
 namespace Nethermind.Network.P2P
 {
-    public class P2PProtocolHandler : ProtocolHandlerBase, IPingSender
+    public class P2PProtocolHandler : ProtocolHandlerBase, IPingSender, IP2PProtocolHandler
     {
         private TaskCompletionSource<Packet> _pongCompletionSource;
         private readonly INodeStatsManager _nodeStatsManager;
         private bool _sentHello;
+        private List<Capability> AgreedCapabilitiesList { get; }
```
Comment (LukaszRozmej) [inline src/Nethermind/Nethermind.Network/P2P/P2PProtocolHandler.cs:38]: wrong naming scheme _agreedCapabilitiesList

**Review by LukaszRozmej — APPROVED** [review]
minor changes, but otherwise good.

**Review by tkstanczak — CHANGES_REQUESTED** [review]
NDM solution got broken, open NDM.sln and fix the build then commit

**Review by tkstanczak — APPROVED** [review]

**Pull request merged**